runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -91,6 +96,20 @@
     @Getter
     private Set<UUID> streamsToIgnore = new HashSet<>();
 
+    /**
+     * We can register streams with non-default type
+     */
+    @Getter
+    private Map<UUID, Class> nonDefaultTypeStreams = new HashMap<>();
```

Comment:
nit: should there be a method "registerStream(streamID, type)" instead of a getter for the entire map?